pandas/core/arrays/string_arrow.py

```diff
@@ -437,18 +437,14 @@ def _str_count(self, pat: str, flags: int = 0):
         return self._convert_int_result(result)
 
     def _str_find(self, sub: str, start: int = 0, end: int | None = None):
-        if start != 0 and end is not None:
-            slices = pc.utf8_slice_codeunits(self._pa_array, start, stop=end)
-            result = pc.find_substring(slices, sub)
-            not_found = pc.equal(result, -1)
-            offset_result = pc.add(result, end - start)
-            result = pc.if_else(not_found, result, offset_result)
-        elif start == 0 and end is None:
-            slices = self._pa_array
-            result = pc.find_substring(slices, sub)
-        else:
+        if (
+            pa_version_under13p0
+            and not (start != 0 and end is not None)
+            and not (start == 0 and end is None)
+        ):
+            # GH#59562
             return super()._str_find(sub, start, end)
-        return self._convert_int_result(result)
+        return ArrowStringArrayMixin._str_find(self, sub, start, end)
```

Comment:
Now that this special case is moved in the mixin method, I would expect this can be removed entirely? (and replaced with a `_str_find = ArrowStringArrayMixin._str_find`)

Reply:
Ah, through `_str_map` using `lib.map_infer_mask`, I suppose. But if there is a cython implementation that is presumably faster, shouldn't we use that for the ArrowDtype as well? 
I saw that in the center PR at https://github.com/pandas-dev/pandas/pull/59624/files#diff-ca6e5560b2fc1721e129b85f10882df8a1f20b5f1ef4dff547170fa35898dfa6R62 you didn't use `_apply_elementwise` but also explicitly went through object dtype. That's for the same reason? Can we use the same pattern?

Reply:
Sure, changed.

Reply:
looks like doing this broke the min_versions build, so reverted